demo/popcha-client/web/src/counterApp.py

```diff
@@ -0,0 +1,68 @@
+"""
+This class contains the function related to the application data persitance
+and  management
+"""
+import secrets
+import time
+
+from flask import render_template
+from werkzeug.datastructures import MultiDict
+
+
+class CounterApp:
+    """
+    An app that allows a user to count
+    """
+
+    def __init__(self):
+        self.app_nonces: dict[str, (str, float)] = {}  # Maps nonce to user
```

Comment:
Then why is there a float? Is a user defined by the pair `(str, float)`? It is quite unclear to me 😅 .